includes/Products/GoogleProductTaxonomy.php

```diff
@@ -0,0 +1,7 @@
+<?php
+namespace SkyVerge\WooCommerce\Facebook\Products;
+// This file was generated using GenerateCategories.php, do not modify it manually
+// php GenerateCategories.php taxonomy-with-ids.en-US.txt
+class GoogleProductTaxonomy {
```

Comment:
Ahh I found the memory usage. When the category data is json encoded in `\SkyVerge\WooCommerce\Facebook\Admin\Google_Product_Category_Field::render` it's using ~100MB of memory on my machine. 

We could use ajax calls instead of encoding the whole array upfront?

Reply:
Interesting. The fb_google_category_to_attribute_mapping.json is something that always was there and I have not touched it. Could you please tell me how exactly have you measured the memory usage - so I could compare what I see?

Reply:
I've been using the Query Monitor plugin measure the memory usage of an entire page load. I've tested the **Product sync** page and I think the same issue can be observed on other other pages where this field is rendered...

**master branch**

<img width="1060" alt="Screen Shot 2021-03-25 at 2 28 52 pm" src="https://user-images.githubusercontent.com/1892137/112416918-857ab180-8d76-11eb-9963-8af52d2cbc4c.png">

**and then this branch**

<img width="1055" alt="Screen Shot 2021-03-25 at 2 28 42 pm" src="https://user-images.githubusercontent.com/1892137/112416937-8e6b8300-8d76-11eb-9b33-6556678f6dcf.png">

Reply:
I could try and use xDebug's profiler tool to analyzer the usage. I tried visually inspecting these changes but I could see exactly where the increase is coming from.

Reply:
Fixed in: 57465095ef860297f41e9fd92a74fafd7eb0f5a4

Reply:
Categories issue fixed in: 96eb2c26116a45d3e05bd768815ab52f63d15718